terralab/logic/pipeline_runs_logic.py

```diff
@@ -93,3 +107,51 @@ def prepare_upload_start_pipeline_run(
     LOGGER.debug(f"Starting {pipeline_name} job {job_id}")
 
     return start_pipeline_run(pipeline_name, job_id, description)
+
+
+## download action
+
+
+def get_result_and_download_pipeline_run_outputs(
+    pipeline_name: str, job_id: uuid.UUID, local_destination: str
+):
+    """Retrieve pipeline run result, download all output files."""
+    LOGGER.info(
+        f"Getting results for {pipeline_name} run {job_id} and downloading to {local_destination}"
+    )
+    result: AsyncPipelineRunResponse = get_pipeline_run_status(pipeline_name, job_id)
+    job_status: str = result.job_report.status
+    LOGGER.debug(f"Job {job_id} status is {job_status}")
+    if job_status != "SUCCEEDED":
+        LOGGER.error(f"Results not available for job {job_id} with status {job_status}")
+        LOGGER.error(handle_non_success(result))
+        exit(1)
+
+    # extract output signed urls and download them all
+    signed_url_list: list[str] = list(result.pipeline_run_report.outputs.values())
+    downloaded_files: list[str] = download_files_with_signed_urls(
+        local_destination, signed_url_list
+    )
+
+    LOGGER.info("All file outputs downloaded:")
+    for local_file_path in downloaded_files:
+        LOGGER.info(indented(local_file_path))
+
+
+def handle_non_success(
+    pipeline_run_response: AsyncPipelineRunResponse,
+) -> Optional[str]:
+    job_status: str = pipeline_run_response.job_report.status
+    if job_status == "SUCCEEDED":
+        return
```

Comment:
this might wanna `raise` if it's meant to handle non success statuses and its getting a success status

Reply:
cool, also renamed the function to be a bit more descriptive